### PR TITLE
Add explicit ids for headers

### PR DESCRIPTION
**Pull request by killall5**

### CODE REVIEW

**Comment by KrishnaPG** [comment]
Wondering if this PR takes care of this situation https://github.com/chjj/marked/issues/664 or falls prey to the same problem.

**Comment by KrishnaPG** [comment]
+1 for this. Guess looks like it resolves the case-inconsistency problem too.

**Comment by Feder1co5oave** [comment]
Please describe what this PR is for.

File: lib/marked.js [hunk 1]
```diff
@@ -146,6 +146,12 @@ Lexer.prototype.lex = function(src) {
  * Lexing
  */
 
+Lexer.prototype.id_generator = function(title, explicit_id) {
+	var draft = explicit_id ? explicit_id : title;
+	draft = this.options.headerPrefix + draft;
+	return draft.toLowerCase().replace(/[^\w]+/g, '-');
```
Comment (KrishnaPG) [inline lib/marked.js]: Explicit ID should be returned as is, without being case-converted or transformed in any way. If author supplied explicit IDs, there must be a good reason why he did so. Hence it should be used as is. No point in transforming it any further.

**Comment by joshbruce** [comment]
Believe this is already taken care of.

**Pull request closed without merging**